controllers/machinedeployment_controller.go

```diff
@@ -136,23 +136,26 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, d *clusterv
 	// This is necessary for CRDs including scale subresources.
 	d.Status.Selector = selector.String()
 
-	// Cluster might be nil as some providers might not require a cluster object
-	// for machine management.
+	// Reconcile and retrieve the Cluster object.
+	if d.Labels == nil {
+		d.Labels = make(map[string]string)
+	}
+	d.Labels[clusterv1.MachineClusterLabelName] = d.Spec.ClusterName
+
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, d.ObjectMeta)
```

Comment:
Same as above here, it seems like we can just use the cluster name to get the cluster here instead of using the labels.